File: src/backend/internal/model/agent.go
package model

import "time"

// Agent 表示可被对话选择和调度的 Agent 配置
type Agent struct {
	ID                    string     `json:"id" db:"id"`
	UserID                *string    `json:"user_id,omitempty" db:"user_id"`
	Name                  string     `json:"name" db:"name"`
	Type                  string     `json:"type" db:"type"`
	CLITool               string     `json:"cli_tool" db:"cli_tool"`
	SystemPrompt          string     `json:"system_prompt,omitempty" db:"system_prompt"`
	ToolsConfig           string     `json:"tools_config,omitempty" db:"tools_config"`
	Avatar                string     `json:"avatar,omitempty" db:"avatar"`
	CapabilitiesJSON      string     `json:"capabilities_json,omitempty" db:"capabilities_json"`
	CustomSkills          string     `json:"custom_skills,omitempty" db:"custom_skills"`
	Tags                  string     `json:"tags,omitempty" db:"tags"`
	Source                string     `json:"source" db:"source"`
	Status                string     `json:"status" db:"status"`
	Version               string     `json:"version,omitempty" db:"version"`
	MachineID             *string    `json:"machine_id,omitempty" db:"machine_id"`
	MachineName           string     `json:"machine_name,omitempty" db:"machine_name"`
	EnableManagementTools bool       `json:"enable_management_tools" db:"enable_management_tools"`
	LastSeenAt            *time.Time `json:"last_seen_at,omitempty" db:"last_seen_at"`
	CreatedAt             time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt             time.Time  `json:"updated_at" db:"updated_at"`
}

// PlatformSkill 是用户在平台 Skill 库中维护、可分配给不同 Agent 的技能。
type PlatformSkill struct {
	ID          string    `json:"id" db:"id"`
	UserID      string    `json:"user_id" db:"user_id"`
	Name        string    `json:"name" db:"name"`
	Description string    `json:"description,omitempty" db:"description"`
	Trigger     string    `json:"trigger,omitempty" db:"trigger"`
	Detail      string    `json:"detail,omitempty" db:"detail"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

File: src/frontend/src/types/agent.ts
export type AgentType = 'system' | 'custom';
export type AgentStatus = 'online' | 'offline' | 'busy' | 'error' | 'stopped';

export interface Agent {
  id: string;
  user_id?: string;
  name: string;
  type: AgentType;
  cli_tool: string;
  system_prompt?: string;
  tools_config?: string;
  avatar?: string;
  capabilities_json?: string;
  custom_skills?: string;
  tags?: string;
  source: string;
  status: AgentStatus;
  version?: string;
  machine_id?: string;
  machine_name?: string;
  enable_management_tools?: boolean;
  last_seen_at?: string;
  created_at: string;
  updated_at: string;
}

export interface PlatformSkill {
  id: string;
  user_id: string;
  name: string;
  category?: string;
  description?: string;
  trigger?: string;
  detail?: string;
  created_at: string;
  updated_at: string;
}

export interface PlatformSkillRequest {
  name: string;
  category?: string;
  description?: string;
  trigger?: string;
  detail?: string;
}

export interface AgentPromptTemplate {
  id: string;
  user_id: string;
  name: string;
  category?: string;
  description?: string;
  system_prompt?: string;
  created_at: string;
  updated_at: string;
}

export interface AgentPromptTemplateRequest {
  name: string;
  category?: string;
  description?: string;
  system_prompt?: string;
}

export interface AgentRequest {
  name: string;
  cli_tool: string;
  system_prompt?: string;
  tools_config?: string;
  avatar?: string;
  capabilities_json?: string;
  enable_management_tools?: boolean;
}

export interface DaemonMachine {
  id: string;
  user_id: string;
  name: string;
  machine_id: string;
  status: 'pending' | 'connected' | 'offline';
  last_seen_at?: string;
  created_at: string;
  updated_at: string;
}

export interface AgentCandidate {
  id: string;
  machine_id: string;
  machine_name: string;
  name: string;
  cli_tool: string;
  version?: string;
  capabilities_json?: string;
  last_seen_at?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateDaemonMachineRequest {
  name: string;
}

export interface CreateDaemonMachineResponse {
  machine: DaemonMachine;
  api_key: string;
  daemon_source_path: string;
  daemon_npm_path: string;
}

export interface AddCandidateAgentRequest {
  name: string;
  cli_tool: string;
  system_prompt?: string;
  tools_config?: string;
  custom_skills?: string;
}

export interface OpenSkillLocationRequest {
  source_path: string;
}
